core/src/main/java/google/registry/tools/ConfirmingCommand.java

```diff
@@ -29,20 +31,46 @@ public abstract class ConfirmingCommand implements Command {
 
   @Override
   public final void run() throws Exception {
-    if (checkExecutionState()) {
-      init();
-      printLineIfNotEmpty(prompt());
-      if (dontRunCommand()) {
-        // This typically happens when all of the work is accomplished inside of prompt(), so do
-        // nothing further.
-        return;
-      } else if (force || promptForYes("Perform this command?")) {
-        System.out.println("Running ... ");
-        System.out.println(execute());
-        printLineIfNotEmpty(postExecute());
-      } else {
-        System.out.println("Command aborted.");
-      }
+    if (!checkExecutionState()) {
+      return;
+    }
+
+    tm().transact(
```

Comment:
I really wish there were a way to make these 3 repeated calls nicer, but I couldn't find any.  You can't extract to a helper function that takes a lambda.